c_src/quicer_listener.c

```diff
@@ -473,13 +474,19 @@ close_listener1(ErlNifEnv *env,
   enif_mutex_lock(l_ctx->lock);
   HQUIC l = l_ctx->Listener;
   l_ctx->Listener = NULL;
+
+  if (l_ctx->is_closed)
+  {
+    ret = ERROR_TUPLE_2(ATOM_CLOSED);
+  }
   l_ctx->is_closed = TRUE;
   enif_mutex_unlock(l_ctx->lock);
 
   // It is safe to close it without holding the lock
   // This also ensures no ongoing listener callbacks
-  // This is a blocking call. @TODO have async version or use dirty scheduler
+  // This is a blocking call.
+  //
   MsQuic->ListenerClose(l);
```

Comment:
is it OK to close the listener if it was already closed? Or should it be better to return just after the critical section if ret != ATOM_OK?

Reply:
It is ok. 
L476 set it to NULL and ListenerClose has `_Pre_defensive_` annotation. So it will never trigger use-after-free.